### ISSUE TITLE
azurerm_application_insights - insights.ComponentsClient#CreateOrUpdate: Failure responding to request: StatusCode=201 Code="Unknown" Message="Unknown service error"

### ISSUE THREAD

**Issue by idelix** [terraform-providers/terraform-provider-azurerm#1771]
Terraform creates azurerm_application_insights  resource but returns error, therefore tf state still see resource as not provisioned preventing successful deployment, this configuration was working fine and just started failing.

### Terraform Version

Terraform v0.11.7
+ provider.azurerm v1.12.0

### Affected Resource(s)

azurerm_application_insights

### Terraform Configuration Files

```
provider "azurerm" {
  subscription_id = "${var.arm_subscription_id}"
  client_id       = "${var.arm_principal}"
  client_secret   = "${var.arm_password}"
  tenant_id       = "${var.tenant_id}"
  alias           = "default"
  version         = "~> 1.8"
}

resource "azurerm_resource_group" "monitor_rg" {
  name     = "aitest-rg"
  location = "NorthEurope"
  provider = "azurerm.default"
}

resource "azurerm_application_insights" "test_ai" {
  name                = "TestNe-ai"
  application_type    = "Other"
  location            = "NorthEurope"
  resource_group_name = "${azurerm_resource_group.monitor_rg.name}"
  provider            = "azurerm.default"
}
```

### Debug Output

terraform apply --var-file=".\terraform.tfvars"

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + azurerm_application_insights.test_ai
      id:                  <computed>
      app_id:              <computed>
      application_type:    "Other"
      instrumentation_key: <computed>
      location:            "northeurope"
      name:                "TestNe-ai"
      resource_group_name: "aitest-rg"
      tags.%:              <computed>

  + azurerm_resource_group.monitor_rg
      id:                  <computed>
      location:            "northeurope"
      name:                "aitest-rg"
      tags.%:              <computed>

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azurerm_resource_group.monitor_rg: Creating...
  location: "" => "northeurope"
  name:     "" => "aitest-rg"
  tags.%:   "" => "<computed>"
azurerm_resource_group.monitor_rg: Creation complete after 1s (ID: /subscriptions/<...>/resourceGroups/aitest-rg)
azurerm_application_insights.test_ai: Creating...
  app_id:              "" => "<computed>"
  application_type:    "" => "Other"
  instrumentation_key: "<sensitive>" => "<sensitive>"
  location:            "" => "northeurope"
  name:                "" => "TestNe-ai"
  resource_group_name: "" => "aitest-rg"
  tags.%:              "" => "<computed>"

Error: Error applying plan:

1 error(s) occurred:

* azurerm_application_insights.test_ai: 1 error(s) occurred:

* azurerm_application_insights.test_ai: insights.ComponentsClient#CreateOrUpdate: Failure responding to request: StatusCode=201 -- Original Error: autorest/azure: Service returned an error. Status=201 Code="Unknown" Message="Unknown service error" Details=[{"etag":"\"<...>\"","id":"/subscriptions/<...>/resourceGroups/aitest-rg/providers/microsoft.insights/components/TestNe-ai","kind":"Other","location":"northeurope","name":"TestNe-ai","properties":{"AppId":"<...>","ApplicationId":"TestNe-ai","Application_Type":"Other","CreationDate":"2018-08-15T10:40:57.6040735+00:00","CustomMetricsOptedInType":null,"Flow_Type":null,"HockeyAppId":null,"HockeyAppToken":null,"InstrumentationKey":"<...>","Name":"TestNe-ai","PackageId":null,"Request_Source":null,"SamplingPercentage":null,"TenantId":"<...>","provisioningState":"Succeeded"},"tags":{},"type":"microsoft.insights/components"}]

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

### Panic Output

* azurerm_application_insights.test_ai: insights.ComponentsClient#CreateOrUpdate: Failure responding to request: StatusCode=201 -- Original Error: autorest/azure: Service returned an error. Status=201 Code="Unknown" Message="Unknown service error"

### Expected Behavior

Terraform should create Appinsight resource and return success

### Actual Behavior

Terraform does create Appinsight resource but returns error

### Steps to Reproduce

Please run provided configuration

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by tombuildsstuff** [comment]
Duplicate of #1661 which has been fixed in #1769

**Comment by tombuildsstuff** [comment]
hi @idelix 

Just to let you know that a fix for this issue has been released as part of v1.13.0 of the AzureRM Provider :)

Thanks!
